public/app/plugins/panel/heatmap-new/fields.ts

```diff
@@ -36,16 +38,23 @@ export interface HeatmapData {
 export function prepareHeatmapData(
   frames: DataFrame[] | undefined,
   options: PanelOptions,
+  timeRange: TimeRange,
   theme: GrafanaTheme2
 ): HeatmapData {
   if (!frames?.length) {
     return {};
   }
 
+  const calcOpts: HeatmapCalculationOptions = {
+    ...options.heatmap,
+    xFrom: timeRange.from.valueOf(),
+    xTo: timeRange.to.valueOf(),
+  };
+
   const { source } = options;
   if (source === HeatmapSourceMode.Calculate) {
     // TODO, check for error etc
-    return getHeatmapData(calculateHeatmapFromData(frames, options.heatmap ?? {}), theme);
+    return getHeatmapData(calculateHeatmapFromData(frames, calcOpts), theme);
```

Comment:
```suggestion
    return getHeatmapData(calculateHeatmapFromData(frames, options.heatmap ?? {}, timeRange), theme);
```
Changing the signature to pass timeRange as a parameter would let us keep the saved options clearly distinct from runtime settings